docs/app/pages/components/components-sections/select/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component } from '@angular/core';
+import { BaseDocumentationSection } from '../../../../../components/base-documentation-section/base-documentation-section';
+import { DocumentationSectionComponent } from '../../../../../decorators/documentation-section-component';
+import { IPlayground } from '../../../../../interfaces/IPlayground';
+import { IPlaygroundProvider } from '../../../../../interfaces/IPlaygroundProvider';
+
+@Component({
+    selector: 'uxd-components-select-custom',
+    templateUrl: './select-custom.component.html',
+    styleUrls: ['./select-custom.component.less']
+})
+@DocumentationSectionComponent('ComponentsSelectCustomComponent')
+export class ComponentsSelectCustomComponent extends BaseDocumentationSection implements IPlaygroundProvider {
+
+    selected: RadioOption;
+    optionList: ReadonlyArray<RadioOption> = [{ name: 'One' }, { name: 'Two' }, { name: 'Three' }, { name: 'Four' }];
+    filter: string = '';
+    showBusyIndicator: boolean = false;
+    allowNull: boolean = false;
+    componentWidth: number = 400;
+    componentMaxHeight: number = 400;
+    placeholder: string = 'Type to search...';
+    focusIndex: number = 0;
+
+    playground: IPlayground = {
+        files: {
+            'app.component.html': this.snippets.raw.appHtml,
+            'app.component.ts': this.snippets.raw.appTs,
+            'app.component.css': this.snippets.raw.appCss
+        },
+        modules: [
+            {
+                imports: ['RadioButtonModule', 'SelectCustomModule'],
+                library: '@ux-aspects/ux-aspects'
+            }
+        ]
+    };
+
+    constructor() {
+        super(require.context('./snippets/', false, /\.(html|css|js|ts)$/));
+    }
+
+    private index(text: string): number {
+        return text.toLowerCase().indexOf(this.filter.toLowerCase());
+    }
+
+    isHidden(name: string): boolean {
```

Comment:
Instead of calling a function to check if an item is hidden, could we update the array whenever the filter text is changed? We try to avoid calling functions from a template as this cause significant slow down on larger pages as the function needs to be called every time change detection is run to check for a change, whereas updating a variable can be checked with a simple equality check.

Reply:
That would mean to maintain two arrays, that's why I preferred to use the function. And especially since this is demo code, I wanted to keep it simple. (By the way, I would need to check if it is contained in the array, not equality.) Therefore, I would prefer to leave it as it is, but I do not have a strong opinion. Does this cause performance issues on the demo page?

Reply:
Yes, it would mean having two arrays, an original list and a visible list. You wouldn't have to check in the view if an item is hidden, You would use a `ReadonlyArray` for the visible items which you update when the query changes. NgFor would bind directly to the list of visible items and handle the visibility for you. Its a case of it might not cause issues on our demo page, but in a page in an enterprise app a simple change like this can make quite a significant improvement in performance which is why we try to avoid function calls in template attributes/inputs wherever possible.

Reply:
I changed an committed it.